source/adapters/cuda/device.cpp

```diff
@@ -1143,17 +1143,18 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceCreateWithNativeHandle(
   if (Result != UR_RESULT_SUCCESS)
     return Result;
 
-  ur_platform_handle_t *Plat = static_cast<ur_platform_handle_t *>(
-      malloc(NumPlatforms * sizeof(ur_platform_handle_t)));
-  Result = urPlatformGet(&AdapterHandle, 1, NumPlatforms, Plat, nullptr);
+  std::vector<ur_platform_handle_t> Platforms(NumPlatforms);
+
+  Result = urPlatformGet(&AdapterHandle, 1, Platforms.size(), Platforms.data(),
+                         nullptr);
```

Comment:
Could `NumPlatforms` be used instead of `Platforms.size()`?

Reply:
we just constructed `Platforms` the line before, and yeah reusing `NumPlatforms` is slightly more efficient than getting the size back out of the vector